packages/@lwc/shared/src/html-attributes.ts

```diff
@@ -170,23 +145,6 @@ export function htmlPropertyToAttribute(propName: string): string {
     return attributeName;
 }
 
-export function htmlAttributeToProperty(attrName: string): string {
-    const ariaPropertyName = AriaAttrNameToPropNameMap[attrName];
-    if (!isUndefined(ariaPropertyName)) {
-        return ariaPropertyName;
-    }
-
-    const specialPropertyName = NO_STANDARD_ATTRIBUTE_PROPERTY_MAPPING.get(attrName);
-    if (!isUndefined(specialPropertyName)) {
-        return specialPropertyName;
-    }
-
-    const cachedPropertyName = CACHED_ATTRIBUTE_PROPERTY_MAPPING.get(attrName);
-    if (!isUndefined(cachedPropertyName)) {
-        return cachedPropertyName;
-    }
-
-    const propertyName = StringReplace.call(attrName, CAMEL_REGEX, (g) => g[1].toUpperCase());
-    CACHED_ATTRIBUTE_PROPERTY_MAPPING.set(attrName, propertyName);
-    return propertyName;
+export function kebabCaseToCamelCase(attrName: string): string {
+    return StringReplace.call(attrName, CAMEL_REGEX, (g) => g[1].toUpperCase());
 }
```

Comment:
See comment above about why I removed this; we want to avoid setting e.g. `ariaLabel` because it's not supported in Firefox, and `ariaLabelledBy` since it's an LWC-specific nonstandard thing, and `tabIndex` because there is some funkiness sometimes with how strings are converted between booleans/numbers/etc. In the template it's an attribute, so it's safer if we just treat it as an attribute.

Reply:
Incidentally without this PR, we are triggering our nonstandard ARIA warnings if you use `aria-labelledby` on an `lwc:external` element. With this PR, the warning goes away.

Reply:
With this change, if the attribute is `aria-readonly`, we check for `ariaReadonly` instead of `ariaReadOnly`, correct?

Reply:
No, we check for the attribute `aria-readonly` rather than any property at all. That's what [this test](packages/@lwc/integration-karma/test/template/directive-external/x/withCamelCaseProperty/withCamelCaseProperty.html) is confirming.